src/codegen/compile.ml

```diff
@@ -3897,6 +3899,19 @@ module IC = struct
       edesc = nr (FuncExport (nr fi))
     })
 
+  let export_timer env =
+    assert !Flags.global_timer;
+    assert (E.mode env = Flags.ICMode || E.mode env = Flags.RefMode);
+    let fi = E.add_fun env "canister_global_timer"
+      (Func.of_body env [] [] (fun env ->
+        G.i (Call (nr (E.built_in env "timer_exp"))) ^^
+        GC.collect_garbage env))
```

Comment:
I guess we'll want to suppress GC here to. Lemme check with Ulan.

Reply:
Right, just like with the heartbeats, timers do not support DTS.

Reply:
@crusso @ggreif 

For the current project that I'm working on, the canister timer will kick off a series of inter-canister calls to retrieve a considerable amount of data and update the state of the canister. This heartbeat initiated "cron job" is the major workhorse of the canister.

@ulan Will heartbeat/timers ever support DTS?

Reply:
@ByronBecker the `job` of an expired timer will always be executed in a context that can do DTS.

Reply:
> Will heartbeat/timers ever support DTS?

Sorry, I missed this question. +1 to Gabor's answer. I am also implement DTS support for the initial heartbeat/global_timer endpoint.